### PR TITLE
Fix batching telemetry thread shutdown

### PR DESCRIPTION
**Pull request by mssalvatore**
# What does this PR do?

BatchingTelemetryMessenger failed to shutdown its thread, causing the agent to hang.

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [x] Is the TravisCI build passing?
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [ ] ~Added relevant unit tests?~
* [ ] Have you successfully tested your changes locally? Elaborate:
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [ ] ~If applicable, add screenshots or log transcripts of the feature working~

### CODE REVIEW

File: monkey/infection_monkey/monkey.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import os
 import subprocess
 import sys
+import threading
```
Comment (mssalvatore) [inline monkey/infection_monkey/monkey.py:6]: Added by mistake

File: monkey/infection_monkey/monkey.py [hunk 2]
```diff
@@ -415,6 +413,12 @@ def cleanup(self):
                 InfectionMonkey._self_delete()
 
         logger.info("Monkey is shutting down")
+        for t in threading.enumerate():
+            logger.info(t.name)
+
+        import faulthandler
+
+        faulthandler.dump_traceback()
```
Comment (mssalvatore) [inline monkey/infection_monkey/monkey.py:421]: Added by mistake

**Comment by mssalvatore** [comment]
@cakekoa Has this been tested yet?

**Comment by cakekoa** [comment]
> @cakekoa Has this been tested yet?

Just tested it. It shuts down cleanly now.

```
2022-08-25 11:50:30,824 [3433:PayloadThread:INFO] automated_master._run_plugins.241: Finished running payloads
2022-08-25 11:50:31,572 [3433:AutomatedMasterThread:DEBUG] automated_master._run_master_thread.90: Waiting for the simulation thread to stop
2022-08-25 11:50:31,572 [3433:MainThread:INFO] automated_master.start.73: The simulation has been shutdown.
2022-08-25 11:50:31,573 [3433:MainThread:INFO] monkey.cleanup.383: Monkey cleanup started
2022-08-25 11:50:33,502 [3433:MonkeyTunnelThread:INFO] tunnel.run.181: Stopping tunnel, waiting for clients: []
2022-08-25 11:50:33,502 [3433:MonkeyTunnelThread:INFO] tunnel.run.198: Closing tunnel
2022-08-25 11:50:36,005 [3433:MainThread:DEBUG] base_telem._log_telem_sending.46: Sending attack telemetry.
2022-08-25 11:50:36,005 [3433:MainThread:DEBUG] base_telem._log_telem_sending.48: Telemetry contents: {"status": 2, "technique": "T1107", "path": "/tmp/monkey_dir_hdcz8yml"}
2022-08-25 11:50:36,273 [3433:MainThread:DEBUG] base_telem._log_telem_sending.46: Sending state telemetry.
2022-08-25 11:50:36,274 [3433:MainThread:DEBUG] base_telem._log_telem_sending.48: Telemetry contents: {"done": true, "version": "1.13.0+dev"}
2022-08-25 11:50:36,290 [3433:MainThread:INFO] monkey.cleanup.414: Monkey is shutting down
$
```

**Pull request merged**